### PR TITLE
add_blocktensor_13

### PR DESCRIPTION
**Pull request by mganahl**
last bit for block sparse tensor support

* fixes some bugs 
* better test coverage for edge cases

### CODE REVIEW

File: tensornetwork/block_sparse/block_tensor.py [hunk 1]
```diff
@@ -1554,6 +1586,15 @@ def tensordot(tensor1: BlockSparseTensor,
         "rank of `tensor2` is smaller than `max(axes2) = {}`".format(
             max(axes1)))
 
+  #inner product
+  if (len(axes1) == tensor1.ndim) and (len(axes2) == tensor2.ndim):
+    return np.dot(
+        tensor1.transpose(axes1, shuffle=True).data,
+        tensor2.transpose(axes2, shuffle=True).data)
+  #outer product
+  if (len(axes1) == 0) and (len(axes2) == 0):
```
Comment (Thenerdstation) [inline tensornetwork/block_sparse/block_tensor.py]: What happens if neither is these are true?

File: tensornetwork/block_sparse/block_tensor.py [hunk 2]
```diff
@@ -2094,8 +2150,10 @@ def trace(tensor: BlockSparseTensor,
       identity = eye(
           Index([out._charges[out._order[i][0]]],
                 [not out._flows[out._order[i][0]]]))
-
+      print(out.data)
```
Comment (Thenerdstation) [inline tensornetwork/block_sparse/block_tensor.py]: Remove print statements

File: tensornetwork/block_sparse/charge_test.py [hunk 3]
```diff
@@ -4,7 +4,7 @@
 from tensornetwork.block_sparse.charge import BaseCharge, intersect, fuse_ndarrays, U1Charge, fuse_degeneracies, fuse_charges
 
 
-def test_BaseCharge_charges():
+def est_BaseCharge_charges():
```
Comment (Thenerdstation) [inline tensornetwork/block_sparse/charge_test.py]: Typo

**Comment by mganahl** [comment]
I need to fix a few more things in the PR. I'll ping you again once its ready!

**Comment by mganahl** [comment]
Hey @Thenerdstation, can you have a look? thanks!
